packages/react-input/src/components/Input/useInput.ts

```diff
@@ -25,21 +25,41 @@ export const inputShorthandProps: (keyof InputSlots)[] = [
  * @param ref - reference to root HTMLInputElement of Input
  */
 export const useInput = (props: InputProps, ref: React.Ref<HTMLElement>): InputState => {
-  const { input, inputWrapper, bookendAfter, bookendBefore, insideEnd, insideStart } = props;
+  const {
+    input,
+    inputWrapper,
+    bookendAfter,
+    bookendBefore,
+    insideEnd,
+    insideStart,
+    size,
+    appearance,
+    inline,
+    ...rest
+  } = props;
+
   return {
+    size,
+    appearance,
+    inline,
     components: {
+      root: 'span',
       input: 'input',
+      inputWrapper: 'span',
+      bookendBefore: 'span',
+      bookendAfter: 'span',
+      insideStart: 'span',
+      insideEnd: 'span',
     },
-    // temporarily must add fake children to prevent getSlots from substituting nullRender
     input: resolveShorthand(input, { required: true }),
     inputWrapper: resolveShorthand(inputWrapper, { required: true }),
     bookendAfter: resolveShorthand(bookendAfter),
     bookendBefore: resolveShorthand(bookendBefore),
     insideEnd: resolveShorthand(insideEnd),
     insideStart: resolveShorthand(insideStart),
-    root: getNativeElementProps('div', {
+    root: getNativeElementProps('span', {
       ref,
-      ...props,
+      ...rest,
```

Comment:
```suggestion
      ...props,
```

Let's keep spreading `props` as it's too easy to loose props with `...rest`. For example is custom handling for `type` in `Button`.

```tsx
const { type, ...rest } = props;
// 💥 Oops, "type" will not go anymore to DOM
const root = getNativeElementProps("div", { ...rest, ref }); 
```

https://github.com/microsoft/fluentui/pull/19483#discussion_r695694564